components/proxy/src/main/java/com/hotels/styx/routing/handlers/StandardHttpPipeline.java

```diff
@@ -77,19 +76,19 @@ public StandardHttpPipeline(List<HttpInterceptor> interceptors, HttpHandler hand
         }
 
         @Override
-        public StyxObservable<HttpResponse> proceed(HttpRequest request) {
+        public Eventual<HttpResponse> proceed(HttpRequest request) {
             if (index < interceptors.size()) {
                 HttpInterceptor.Chain chain = new HttpInterceptorChain(this, index + 1);
                 HttpInterceptor interceptor = interceptors.get(index);
 
                 try {
                     return interceptor.intercept(request, chain);
                 } catch (Throwable e) {
-                    return StyxObservable.error(e);
+                    return Eventual.error(e);
                 }
             }
-            return fromRxObservable(toRxObservable(client.handle(request, this.context))
-                    .compose(StandardHttpPipeline::sendErrorOnDoubleSubscription));
+            return new Eventual<>(RxReactiveStreams.toPublisher(RxReactiveStreams.toObservable(client.handle(request, this.context))
```

Comment:
Looks like these two `RxReactiveStreams` methods can be statically-imported. It would also be nice to lay out the code in a way that makes it more clear why we turn a publisher to an observable and then back again.